terraform_providers_lock.sh

```diff
@@ -2,33 +2,53 @@
 
 set -eo pipefail
 
-main() {
-  initialize_
-  parse_cmdline_ "$@"
-  terraform_providers_lock_
+function main {
+  common::initialize
+  common::parse_cmdline "$@"
+  common::per_dir_hook "${ARGS[*]}" "${FILES[@]}"
 }
 
-initialize_() {
+function common::colorify {
+  # Colors. Provided as first string to first arg of function.
+  # shellcheck disable=SC2034
+  local -r red="$(tput setaf 1)"
+  # shellcheck disable=SC2034
+  local -r green="$(tput setaf 2)"
+  # shellcheck disable=SC2034
+  local -r yellow="$(tput setaf 3)"
+  # Color reset
+  local -r RESET="$(tput sgr0)"
```

Comment:
Why using `tput` (which btw behaves differently on different BSD systems) and no just escape-codes?
```suggestion
  local -r red="\e[0m\e[31m"
  # shellcheck disable=SC2034
  local -r green="\e[0m\e[32m"
  # shellcheck disable=SC2034
  local -r yellow="\e[0m\e[33m"
  # Color reset
  local -r RESET="\e[0m"
```

Reply:
Hmm, are you sure that `tput` will not provide the same colors for macOS and Linux?
I'm not a fan of escape sequences.

Reply:
I don't have MacOS, though on FreeBSD `tput {setaf,sgr0}` doesn't do what it's expected to comparing with Ubuntu, whereas escape codes work as intended on both.

Reply:
Well, if we support FreeBSD, we totally should start supporting Windows :D 

OK, I will change it to escape codes

Reply:
This is not literally about supporting different OS'es, but about compatibility and portability across different *NIX implementations, including e.g. difference in POSIX and GNU tools and their derivatives.